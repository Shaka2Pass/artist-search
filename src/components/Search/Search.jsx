//Search by Artist name returns array of artist with similar name by score.
//Allow paging through artists Check 4/30 or 5/1 pagination. 
//http://musicbrainz.org/ws/2/artist?query=<SEARCH>&fmt=json&limit=25
// Search should display a list of matching artists
// Use `http://musicbrainz.org/ws/2/artist?query=<SEARCH>&fmt=json&limit=25`
// * BONUS: As you type into the input box update a `searchTerm`query string.
//   Navigating directly to a URL with a `searchTerm` query string pre-fills
//   the search form.

import React from 'React';
import PropTypes from 'prop-types';

const Search = ({ text, onSubmit, artistName = 'text', onChange }) => (
  <form onSubmit={onSubmit}>
    <input type="text" name={artistName} value={text} onChange={onChange} />
    <button>Search</button>
  </form>
);

Search.propTypes = {
  text: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  artistName: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Search;



 
 
