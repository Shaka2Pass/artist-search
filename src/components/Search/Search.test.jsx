import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('It describes the search component', () => {
  it('matches a snapshot of Search.jsx', () => {
    const wrapper = shallow(<Search 
      text="U2"
      onSubmit={()=>{}}
      artistName="U2"
      onChange={()=>{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});



