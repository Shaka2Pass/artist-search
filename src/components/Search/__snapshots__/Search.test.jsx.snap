// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It describes the search component matches a snapshot of Search.jsx 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="U2"
    onChange={[Function]}
    type="text"
    value="U2"
  />
  <button>
    Search
  </button>
</form>
`;
